### PR TITLE
SASS-1026 : Added Stub route for declareCrystallisationForTaxYear

### PR DESCRIPTION
**Pull request by robert-webb-hmrc**
### Description
Added Stub route for declareCrystallisationForTaxYear

Add a link to the relevant story in Jira
[SASS-1026](https://jira.tools.tax.service.gov.uk/browse/SASS-865)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by robert-webb-hmrc** [comment]
Please review and merge SASS-1028 first

File: conf/prod.routes [hunk 1]
```diff
@@ -12,6 +12,9 @@ GET     /income-tax/income-sources/nino/:nino
 ### DES #1393 - v1.2.0 ###
 POST    /income-tax/income-sources/nino/:nino                                   controllers.IncomeSourcesController.createIncomeSource(nino:String)
 
+### DES #1416 - v1.0.0 ###
+POST    /income-tax/calculation/:idType/:nino/:taxYear/:calculationId/declare-crystallisation     controllers.CalculationController.declareCrystallisationForTaxYear(idType: String, nino: String, taxYear: Int, calculationId: String)
```
Comment (dnyuar) [inline conf/prod.routes:16]: Shouldn't the idType always be a nino?

File: app/controllers/CalculationController.scala [hunk 2]
```diff
@@ -41,4 +41,14 @@ class CalculationController @Inject()(cc: ControllerComponents) extends BackendC
     }
   }
 
+  // DES #1416 - v1.0.0 //
+  def declareCrystallisationForTaxYear(idType: String, nino: String, taxYear: Int, calculationId: String): Action[JsValue] = Action.async(parse.json) { implicit request =>
+    if (request.body != Json.parse("{}")) {
+      val message = s"[declareCrystallisationForTaxYear] API needs empty json supplied. Nino with request: $nino"
```
Comment (dnyuar) [inline app/controllers/CalculationController.scala:47]: taxYear and calculationId could be passed into the message

File: app/controllers/CalculationController.scala [hunk 3]
```diff
@@ -41,4 +41,14 @@ class CalculationController @Inject()(cc: ControllerComponents) extends BackendC
     }
   }
 
+  // DES #1416 - v1.0.0 //
+  def declareCrystallisationForTaxYear(idType: String, nino: String, taxYear: Int, calculationId: String): Action[JsValue] = Action.async(parse.json) { implicit request =>
```
Comment (dnyuar) [inline app/controllers/CalculationController.scala:45]: Shouldn't the idType always be a nino?

**Review by dnyuar — APPROVED** [review]

File: conf/prod.routes [hunk 4]
```diff
@@ -12,6 +12,9 @@ GET     /income-tax/income-sources/nino/:nino
 ### DES #1393 - v1.2.0 ###
 POST    /income-tax/income-sources/nino/:nino                                   controllers.IncomeSourcesController.createIncomeSource(nino:String)
 
+### DES #1416 - v1.0.0 ###
+POST    /income-tax/calculation/nino/:nino/:taxYear/:calculationId/declare-crystallisation     controllers.CalculationController.declareCrystallisationForTaxYear(nino: String, taxYear: Int, calculationId: String)
```
Comment (BenCosfordHMRC) [inline conf/prod.routes:16]: declare-crystallisation is our endpoint, with this being the DES one it should be /crystallise at the end instead

Comment (dnyuar) [reply conf/prod.routes:16]: You're right, i'm not sure how I missed that

**Pull request merged**